tools/telemetry-generator/src/handlers/customBenchmarkHandler.ts

```diff
@@ -0,0 +1,25 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+module.exports = function handler(fileData, logger): void {
+	if (process.env.FLUID_ENDPOINTNAME === undefined) {
+		console.log("ENDPOINTNAME not defined using local as default.");
+	} else {
+		console.log("ENDPOINTNAME", process.env.FLUID_ENDPOINTNAME);
+	}
+
+	for (const testData of fileData.benchmarks) {
+		logger.send({
+			namespace: "FFEngineering", // Transfer the telemetry associated with tests performance measurement to namespace "FFEngineering"
+			category: "performance",
+			driverEndpointName: process.env.FLUID_ENDPOINTNAME ?? "",
+			eventName: "Benchmark",
+			benchmarkType: "CustomBenchmark",
+			suiteName: fileData.suiteName,
+			benchmarkName: testData.benchmarkName,
+			...testData.customData,
```

Comment:
I realized that the handlers for appInsights and non-appInsights are slightly different. [AppInsights](https://github.com/microsoft/FluidFramework/blob/main/tools/telemetry-generator/src/handlers/appInsightsExecutionTimeTestHandler.ts#L16) separately logs values where as [non-AppInsights](https://github.com/microsoft/FluidFramework/blob/main/tools/telemetry-generator/src/handlers/executionTimeTestHandler.ts) sends all the values in a batch. 

This is why I simply used `...` spreader and avoided using nested for-loop to access the `customData`. Anyways, why are we more _cautious_ (considering the usage of try-catch) of AppInsights handler potentially not reading the output files?

Reply:
Spreader here seems fine. Minor concern about it potentially overriding one of the other properties we put in the event, but since we own the keys in custom data, probably not a big concern.

The try-catch doesn't really cover issues reading the output files (the contents of the file are expected in the `fileData` parameter. But it protects us from potential network issues, and it's particularly important there precisely because we have to issue a separate call/request for each metric. If the first one failed and didn't catch, none of the others would be sent. Whereas in the non-app-insights handler, since it's all a single request, we _could_ try-catch, but then there'd be nothing interesting to do in the catch other than maybe retrying (but I believe the logger might already have some retry capabilities).